src/cmd-buildextend-squashfs

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/python3 -u
+# An operation that mutates a build by generating a squashfs rootfs.
+
+import os
+import sys
+import json
+import yaml
+import shutil
+import argparse
+
+sys.path.insert(0, '/usr/lib/coreos-assembler')
+from cmdlib import run_supermin, run_verbose, write_json, sha256sum_file
+
+# Parse args and dispatch
+parser = argparse.ArgumentParser()
+parser.add_argument("--build", help="Build ID")
+args = parser.parse_args()
+
+# default to latest build if not specified
+if not args.build:
+    with open('builds/builds.json') as f:
+        j = json.load(f)
+        args.build = j['builds'][0]
+
+print(f"Targeting build: {args.build}")
+
+with open('src/config/manifest.yaml') as f:
+    manifest = yaml.safe_load(f)
+
+base_name = manifest['rojig']['name']
+img_prefix = f'{base_name}-{args.build}'
+artifact_name = f'{img_prefix}.rootfs.squashfs'
+
+builddir = f'builds/{args.build}'
+buildmeta_path = f'{builddir}/meta.json'
+with open(buildmeta_path) as f:
+    buildmeta = json.load(f)
+
+tmpdir = 'tmp/buildpost-squashfs'
+if os.path.isdir(tmpdir):
+    shutil.rmtree(tmpdir)
+os.mkdir(tmpdir)
+
+
+def generate_squashfs(workdir):
+    ref = buildmeta['ref']
+    tmp_img = f'{tmpdir}/{artifact_name}'
+    script = f'''#!/usr/bin/env bash
+set -xeuo pipefail
+ostree checkout --repo {workdir}/repo/ {ref} /tmp/rootfs
+/sbin/mksquashfs /tmp/rootfs {workdir}/{tmp_img}
```

Comment:
This is just a stub right?  The commit description says it's bootable this won't do that AFAICS.

This whole concept kind of depends on us reworking to use not-anaconda right?

Reply:
Yes and no.

This a basic ostree->squashfs conversion, and the resulting artifact can already be run by `systemd-nspawn --volatile --boot` (I wouldn't say it's bootable, as it doesn't contain any bootloader). No very interesting per se, but (possibly) an intermediate step towards re-using this rootfs for embeddeding in live PXE/ISO artifacts.

In that regard, this flow is already bypassing anaconda. It consumes an ostree ref and runs this build-extending script in a supermin VM.